apps/src/lab2/levelEditors/predictSettings/EditPredictSettings.tsx

```diff
@@ -0,0 +1,119 @@
+import React, {useState} from 'react';
+import {LevelPredictSettings, PredictQuestionType} from '../types';
+import {SimpleDropdown} from '@cdo/apps/componentLibrary/dropdown';
+import Checkbox from '@cdo/apps/componentLibrary/checkbox';
+import moduleStyles from './edit-predict-settings.module.scss';
+import {BodyThreeText} from '@cdo/apps/componentLibrary/typography';
+import FreeResponseFields from './FreeResponseFields';
+import MultipleChoiceFields from './MultipleChoiceFields';
+
+interface EditPredictSettingsProps {
+  initialSettings: LevelPredictSettings;
+}
+
+const EditPredictSettings: React.FunctionComponent<
+  EditPredictSettingsProps
+> = ({initialSettings}) => {
+  const [predictSettings, setPredictSettings] = useState<LevelPredictSettings>(
+    initialSettings || {is_predict_level: false}
+  );
+
+  const handleIsPredictToggle = () => {
+    setPredictSettings({
+      ...predictSettings,
+      is_predict_level: !predictSettings.is_predict_level,
+      question_type:
+        predictSettings.question_type || PredictQuestionType.FreeResponse,
+    });
+  };
+
+  const handleQuestionTypeChange = (
```

Comment:
Looks like we don't clear out settings when changing the question type, right? I can see that being useful so work isn't lost when toggling between types, but when saving to level, should we clear out unused data? Any potential issues if a level has both free response data and multiple choice data?

Reply:
Yeah I debated this, it's probably a good idea to clear out the data on save. I was planning to ignore unused fields on the frontend but it's definitely cleaner to just delete them.